### PR TITLE
Add datanode cluster metrics and dashboard

### PR DESCRIPTION
**Pull request by moesterheld**
/nocl

/prd Graylog2/graylog-plugin-enterprise#6478

## Description
Records data node cluster metrics to metrics data stream and provides the dashboard to show them

**UI**: if you have a valid Enterprise license and Datanode is configured, Cluster Management Dynamic Search will be displayed. Otherwise we just display a Static Teaser Search.

Screenshot 2024-02-06 at 14 14 06

## Motivation and Context
fixes https://github.com/Graylog2/graylog2-server/issues/17704

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: changelog/unreleased/issue-17704.toml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+type = "fixed"
```
Comment (grotlue) [inline changelog/unreleased/issue-17704.toml:1]: It's a new feature, no bugfix, right?

File: graylog2-web-interface/src/components/datanode/ClusterManagement/ClusterManagementTeaserSearch.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+import viewJson from './sample-dashboards/overview_view.json';
+import searchJson from './sample-dashboards/overview_search.json';
+import resultJson from './sample-dashboards/overview_results.json';
+
+const hotspots = [
+  {
+    positionX: '50%',
+    positionY: '120px',
+    description: 'Show performance metrics of your data node and managed OpenSearch cluster.',
```
Comment (grotlue) [inline graylog2-web-interface/src/components/datanode/ClusterManagement/ClusterManagementTeaserSearch.tsx:29]:
```suggestion
    description: 'Show performance metrics of your Data Node and managed OpenSearch cluster.',
```

**Review by grotlue — CHANGES_REQUESTED** [review]
While testing I came across some points which I find confusion and I think the user would too:

❌ The footer (showing the current Graylog version) is shown twice

❌ The dashboard title is `datanodes`, the in the summary and description it's data nodes. It;s stick to Data Nodes.
❓Does the dashboard title, summary and description need to be editable?
❓Is it necessary that the user can mark the dashboard as favorite, even though it does not have any real functionality?
❌ Some buttons are not usable, e.g. widget edit button. When I click them they redirect to "empty pages" with is a weird behaviour. When I click the browser back button on those empty pages I am back on the *Data Nodes* tab, not *Cluster Management*
❌ It is possible to save this dashboard as a new dashboard using the same title. It will then also appear in the dashboard list as usual. However, obviously it's a different dashboard, so going back to the DataNode Cluster Metrics I see the "old" version of the dashboard - at least that is how the user could perceive it.

**Review by grotlue — APPROVED** [review]
Looking good, thank you for the changes!

**Review by todvora — APPROVED** [review]
LGTM, awesome work!

**Pull request merged**